src/plugin.ts

```diff
@@ -260,12 +260,15 @@ export class MongoDBOIDCPluginImpl implements MongoDBOIDCPlugin {
     let issuer = new Issuer(issuerParams);
     let client = new issuer.Client(clientParams);
 
+    const oidcStateParam = (await promisify(randomBytes)(16)).toString('hex');
     const server = new RFC8252HTTPServer({
       redirectUrl: this.getRedirectURI(),
       logger: this.logger,
+      redirectServerRequestHandler: this.options.redirectServerRequestHandler,
+      oidcStateParam,
```

Comment:
Just in case you’re reviewing files in Github’s alphabetical order, the `oidcStateParam` changes are explained in a comment in `RFC8252HTTPServer` (i.e.: these changes aren’t strictly necessary for security/auth but allow for much nicer error handling)